types/host.go

```diff
@@ -82,6 +82,16 @@ func (host HostInfo) Uptime() time.Duration {
 	return time.Since(host.BootTime)
 }
 
+// FQDNAwareHostname returns the system hostname, honoring the given
+// flag to report the FQDN or not.
+func (host HostInfo) FQDNAwareHostname(wantFQDN bool) string {
```

Comment:
Why should this become part of the go-sysinfo API? What about a helper function somewhere outside of go-sysinfo for this?

Reply:
The helper seemed generally useful enough (to me) for anyone asking for the system hostname in an FQDN-aware fashion.  But if you feel strongly that this pollutes the go-sysinfo API then I'm happy to make a similar helper in elastic-agent, where this is being used as of now.

Reply:
Yeah, if you wouldn't mind moving it there I would appreciate it. This package has a pretty simple exported API and this helper, while simple, seems pretty specific to the feature flag use case.